cpp/src/visitors/graph_make_visitor.cpp

```diff
@@ -0,0 +1,164 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cugraph/algorithms.hpp>
+#include <cugraph/visitors/graph_make_visitor.hpp>
+
+namespace cugraph {
+namespace visitors {
+//
+// wrapper code:
+//
+template <typename vertex_t, typename edge_t, typename weight_t, bool st, bool mg>
+void graph_maker_visitor<vertex_t,
+                         edge_t,
+                         weight_t,
+                         st,
+                         mg,
+                         std::enable_if_t<is_candidate<vertex_t, edge_t, weight_t>::value>>::
+  visit_graph(graph_envelope_t::base_graph_t const& graph)
+{
+  if constexpr (mg == false) {
+    auto const& v_args = ep_.get_args();
+
+    auto num_args = v_args.size();
+    assert(num_args > 6);
+
+    // cnstr. args unpacking:
+    //
+    raft::handle_t* ptr_handle = static_cast<raft::handle_t*>(v_args[0]);
+    vertex_t const* p_src      = static_cast<vertex_t*>(v_args[1]);
+    vertex_t const* p_dst      = static_cast<vertex_t*>(v_args[2]);
+    weight_t const* p_weights  = static_cast<weight_t*>(v_args[3]);
+    edge_t num_edges           = *static_cast<edge_t*>(v_args[4]);
+    vertex_t num_vertices      = *static_cast<vertex_t*>(v_args[5]);
+    bool check                 = *static_cast<bool*>(v_args[6]);
+
+    bool is_sym{false};
+    bool is_multigraph{false};
+
+    if (num_args > 7) {
+      is_sym = *static_cast<bool*>(v_args[7]);
+      if (num_args > 8) is_multigraph = *static_cast<bool*>(v_args[8]);
+    }
+
+    cugraph::graph_properties_t graph_props{is_sym, is_multigraph};
+
+    std::optional<weight_t const*> opt_ptr_w;
+    if (p_weights) { opt_ptr_w = p_weights; }
+
+    cugraph::edgelist_t<vertex_t, edge_t, weight_t> edgelist{p_src, p_dst, opt_ptr_w, num_edges};
+
+    cugraph::graph_meta_t<vertex_t, edge_t, false> meta{num_vertices, graph_props, std::nullopt};
+    erased_pack_t ep_graph{ptr_handle, &edgelist, &meta, &check};
+
+    DTypes vertex_tid = reverse_dmap_t<vertex_t>::type_id;
+    DTypes edge_tid   = reverse_dmap_t<edge_t>::type_id;
+    DTypes weight_tid = reverse_dmap_t<weight_t>::type_id;
+    bool st_id        = st;
+    bool mg_id        = mg;
+    GTypes graph_tid  = GTypes::GRAPH_T;
+
+    graph_envelope_t graph_envelope{
+      vertex_tid, edge_tid, weight_tid, st_id, mg_id, graph_tid, ep_graph};
+
+    result_ = return_t{std::move(graph_envelope)};
+  } else {
+    CUGRAPH_FAIL("Graph factory visitor not currently supported (multi_gpu == true).");
+  }
+}
+
+// EIDir's:
+//
+template class graph_maker_visitor<int, int, float, true, true>;
```

Comment:
In the spirit of keeping PRs short and targetted I'd rather not handle this piece of functionality in this PR. It's unrelated and would trigger additional levels of separation (I cannot just divorce different SFINAEs of the same class into separate files without additional intermediary headers in between, etc.) Again, this is a lateral goal, separate from this PR. I can address that change in bulk afterwards.

Reply:
Certainly a lateral goal.  Certainly the change I reference here (splitting things into separate files) happened after you began work on this.  I'm OK deferring in the interest of getting this PR merged.

Generally, I think if a PR is open for a long time and things change in the baseline that we should reflect those changes into the PR rather than add to the backlog of unfinished work.  Agree that this falls into the gray area since the MG/SG separation is still an incomplete task and this step can easily be added to either the next PR on this topic or the next MG/SG separation PR.  But if the MG/SG separation task had been completed during this time window then merging this PR could potentially have broken things.